subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -493,39 +472,90 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
 
 static bool broadcast_source_found(struct bt_data *data, void *user_data)
 {
+	static char identified_addr_str[BT_ADDR_LE_STR_LEN];
+	static uint32_t broadcast_id = INVALID_BROADCAST_ID;
 	struct bt_bap_broadcast_assistant_add_src_param param = { 0 };
 	const struct bt_le_scan_recv_info *info = user_data;
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
-	uint32_t broadcast_id;
+	bool identified_broadcast = false;
 	int err;
 
 	/* Verify that it is a BAP broadcaster*/
 
-	if (data->type != BT_DATA_SVC_DATA16) {
-		return true;
-	}
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
 
-	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
-		return true;
-	}
+	switch (data->type) {
+	case BT_DATA_SVC_DATA16:
+		if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+			return true;
+		}
 
-	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
-		return true;
-	}
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		auto_scan.identified_id = true;
+
+		if (broadcast_id == auto_scan.broadcast_id) {
+			identified_broadcast = true;
+		}
+
+		if (auto_scan.identified_name) {
+			if (!strncmp(identified_addr_str, addr_str, strlen(addr_str))) {
+				identified_broadcast = true;
+			}
+		} else {
+			bt_addr_le_to_str(info->addr, identified_addr_str,
+					sizeof(identified_addr_str));
+		}
+
+		break;
+	case BT_DATA_BROADCAST_NAME:
+		if (strlen(auto_scan.broadcast_name) == 0U) {
+			return true;
+		}
+
+		if (!IN_RANGE(data->data_len, BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+					BT_AUDIO_BROADCAST_NAME_LEN_MAX)) {
```

Comment:
Indentation is off (in Zephyr tabs = 8 spaces)

Reply:
Corrected.